lib/vrf.c

```diff
@@ -994,25 +994,50 @@ const char *vrf_get_default_name(void)
 	return vrf_default_name;
 }
 
-int vrf_bind(vrf_id_t vrf_id, int fd, const char *name)
+int vrf_bind(vrf_id_t vrf_id, int fd, const char *ifname)
 {
 	int ret = 0;
 	struct interface *ifp;
+	struct vrf *vrf;
 
-	if (fd < 0 || name == NULL)
+	if (fd < 0)
 		return fd;
-	/* the device should exist
-	 * otherwise we should return
-	 * case ifname = vrf in netns mode => return
-	 */
-	ifp = if_lookup_by_name(name, vrf_id);
-	if (!ifp)
+
+	if (vrf_id == VRF_UNKNOWN)
 		return fd;
```

Comment:
I think this is really bad semantics. I know this is how it was before, but since we've established a convention of returning < 0 in error case on line 1004 it's totally wack to return a positive value here. And I bet most callers don't check the return value against the FD they passed as an error case. What do you think about changing this to -1 and saying as much in a comment over the function declaration in the header?

Reply:
I agree. Fixed this and other returns. Now the function always returns 0 on success and -1 on failure.